File: src/main/java/com/personal/projects/quiz/services/ParticipantAnswerService.java
package com.personal.projects.quiz.services;

import com.personal.projects.quiz.models.ParticipantAnswer;

import java.util.List;
import java.util.Optional;

public interface ParticipantAnswerService {

    ParticipantAnswer save(ParticipantAnswer participantAnswer);
    List<ParticipantAnswer> findByQuizIdAndParticipantId(Integer quizId, Integer participantId);
}

File: frontend/pet-store/src/app/services/participantAnswer.service.ts
import { Injectable } from '@angular/core';
import { Observable } from "rxjs";
import { HttpClient } from "@angular/common/http";
import { ParticipantAnswer } from "../models/participantAnswer";

@Injectable({
  providedIn: 'root'
})
export class ParticipantAnswerService {

  constructor(private http: HttpClient) { }

  save(participantAnswer: ParticipantAnswer): Observable<ParticipantAnswer> {
    return this.http.post<ParticipantAnswer>('http://localhost:9000/participant-answer/save', participantAnswer);
  }

  get(quizId, participantId: number): Observable<ParticipantAnswer[]> {
    return this.http.get<ParticipantAnswer[]>('http://localhost:9000/participant-answer/quiz/'+quizId+'/participant/'+participantId);
  }

}
